docs/source/tracking/artifacts-stores.rst

```diff
@@ -0,0 +1,200 @@
+how to connect to the remote storage.
+
+.. _artifacts-stores-manage-access:
+
+Manage Access
+-------------
+To allow the server and clients to access the artifact location, you should configure your cloud
+provider credentials as normal. For example, for S3, you can set the ``AWS_ACCESS_KEY_ID``
+and ``AWS_SECRET_ACCESS_KEY`` environment variables, use an IAM role, or configure a default
+profile in ``~/.aws/credentials``.
+
+.. important::
+    Access credentials and configuration for the artifact storage location are configured *once during server initialization* in the place
+    of having users handle access credentials for artifact-based operations. Note that *all users who have access to the
+    Tracking Server in this mode will have access to artifacts served through this assumed role*.
+
+Timeout
+-------
+You can set an environment variable ``MLFLOW_ARTIFACT_UPLOAD_DOWNLOAD_TIMEOUT`` (in seconds) to configure the timeout for artifact uploads and downloads.
+Note that this is experimental feature, may be changed or removed.
+
+Logging Artifacts Location
+--------------------------
+MLflow automatically records ``artifact_uri`` property as a part of :py:class:`mlflow.entities.RunInfo`, so you can
+retrieve the location of the artifacts for historical runs using the :py:func:`mlflow.get_artifact_uri` API. 
+Also, ``artifact_location`` is a property recorded on :py:class:`mlflow.entities.Experiment` for
+default location to store artifacts for all runs in this experiment.
+
+.. important::
+
+  If you do not specify a ``--default-artifact-root`` or an artifact URI when creating the experiment
+  (for example, ``mlflow experiments create --artifact-location s3://<my-bucket>``), the artifact root
+  is a path inside the file store. Typically this is not an appropriate location, as the client and
+  server probably refer to different physical locations (that is, the same path on different disks).
+
+.. _artifacts-store-supported-storages:
+
+Supported storages
+==================
+
+Amazon S3 and S3-compatible storage
+-----------------------------------
+
+To store artifacts in S3 (whether on Amazon S3 or on an S3-compatible alternative, such as
+`MinIO <https://min.io/>`_ or `Digital Ocean Spaces <https://www.digitalocean.com/products/spaces>`_), specify a URI of the form ``s3://<bucket>/<path>``. MLflow obtains
+credentials to access S3 from your machine's IAM role, a profile in ``~/.aws/credentials``, or
+the environment variables ``AWS_ACCESS_KEY_ID`` and ``AWS_SECRET_ACCESS_KEY`` depending on which of
+these are available. For more information on how to set credentials, see
+`Set up AWS Credentials and Region for Development <https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/setup-credentials.html>`_.
+
+To add S3 file upload extra arguments, set ``MLFLOW_S3_UPLOAD_EXTRA_ARGS`` to a JSON object of key/value pairs.
```

Comment:
For searchability, we might want to wrap each of these examples and their explanation introductions in an h3 header ``^^^^``. These examples are pretty important.

Reply:
True, will make them distinct subsections.